app/charts/map/map.tsx

```diff
@@ -114,21 +130,27 @@ export const MapComponent = () => {
     identicalLayerComponentIris,
     areaLayer,
     symbolLayer,
+    bounds,
     bbox,
   } = useChartState() as MapState;
+  const isViewStateLocked = controlsType === "locked";
+  const initialViewState = useMemo(() => {
+    return !isViewStateLocked
+      ? BASE_VIEW_STATE
+      : getInitialViewState(bbox, bounds);
+  }, [isViewStateLocked, bbox, bounds]);
   const locale = useLocale();
 
   const [, dispatchInteraction] = useInteraction();
   const [, setMapTooltipType] = useMapTooltip();
 
-  const [viewState, setViewState] = useState(INITIAL_VIEW_STATE);
+  const [viewState, setViewState] = useState(initialViewState);
```

Comment:
We don't need a memo if we use the value only for the initial value of the state. Pass directly the useMemoed function to useState in this case.